typed/remove.go

```diff
@@ -92,6 +95,11 @@ func (w *removingWalker) doMap(t *schema.Map) ValidationErrors {
 	m.Iterate(func(k string, val value.Value) bool {
 		pe := fieldpath.PathElement{FieldName: &k}
 		path, _ := fieldpath.MakePath(pe)
+
+		if w.toRemove.Has(path) {
+			return true
+		}
+
```

Comment:
Line 107 below, we're doing another similar test that I think you can remove.

Reply:
I'm also very surprised that you don't have to do that for lists (in case we want to remove an associative list-item for example). I guess I'll see in the tests if that's a use-case you missed.

Reply:
I think i missed a test case for this.

Reply:
Added a test, but list already does this.